fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -306,16 +306,25 @@ private void addNotYetAddedVehicles(long time) throws InternalFederateException
             int departIndex = vehicleRegistration.getDeparture().getDepartureConnectionIndex();
             String departSpeed = extractDepartureSpeed(vehicleRegistration);
             String laneId = extractDepartureLane(vehicleRegistration);
+            if (externalVehicleMap.containsKey(vehicleId)) {
+                if (externalVehicleMap.get(vehicleId).isAdded()) {
+                    iterator.remove();
+                    continue;
+                }
+                // TODO: Find better solution. Currently, an arbitrary SUMO route for external vehicles is selected, since a registered
+                //       SUMO route is required when adding a vehicle to SUMO. Using an empty route id "" leads to an error.
+                Object[] routeIds = routeCache.keySet().toArray();
+                routeId = (String) routeIds[0];
```

Comment:
better use `routeId = Iterables.getFirst(routeCache.keySet(), null);`
This way you also do not get a `IndexOutOfBoundsException` if no route is known.